security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthManager.scala

```diff
@@ -0,0 +1,162 @@
+package pl.touk.nussknacker.ui.security.api
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.{
+  AuthorizationFailedRejection,
+  Directive0,
+  Directive1,
+  Directives,
+  RejectionHandler,
+  Route
+}
+import akka.http.scaladsl.server.Directives.{complete, handleRejections, optionalHeaderValueByName, provide, reject}
+import pl.touk.nussknacker.security.AuthCredentials.{
+  ImpersonatedAuthCredentials,
+  NoCredentialsProvided,
+  PassedAuthCredentials
+}
+import pl.touk.nussknacker.security.{AuthCredentials, ImpersonatedUserIdentity}
+import pl.touk.nussknacker.ui.security.api.AuthManager.{ImpersonationConsideringInputEndpoint, impersonateHeaderName}
+import pl.touk.nussknacker.ui.security.api.CreationError.ImpersonationNotAllowed
+import pl.touk.nussknacker.ui.security.api.SecurityError.{
+  BaseAuthenticationError,
+  BaseAuthorizationError,
+  ImpersonatedUserDataNotFoundError,
+  ImpersonationMissingPermissionError
+}
+import sttp.tapir._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AuthManager(authenticationResources: AuthenticationResources)(implicit ec: ExecutionContext) {
```

Comment:
WDYT about creating two traits `AkkaBasedAuthManager` and moving there the code related to the akka things? In the future, we will just remove it. At the moment there are not so many (non-trivial) common code lines in this class. Moreover, we are not able to easily make these two cases symmetric. Reading the code of the AuthManager is not easy because akka and trapir things are mixed. Maybe it's worth to exclude the akka things to the private trait (for a sake of easy getting rid of it in the future).